### ISSUE TITLE
Make debug loggers throughout monorepo uniform in namespacing

### ISSUE THREAD

**Issue by scorbajio** [ethereumjs/ethereumjs-monorepo#3689]
This issue is inspired by incongruities discovered from work done on #3663 and #3676. 

The two issues that need addressing include the following:
1. [Debug namespaces are currently named using a mix of casings](https://github.com/ethereumjs/ethereumjs-monorepo/pull/3663#discussion_r1759257225)
2. [Debug namespaces currently don't allow a parent namespace to be included when a wild card is used for including it's subspaces. E.g. `DEBUG=ethjs,blockchain:*` will include `blockchain:clique` but not `blockchain`.](https://github.com/ethereumjs/ethereumjs-monorepo/pull/3676#discussion_r1765788542)

Another issue I want to note is making sure that any additional debug logic is also being guarded with `this.DEBUG` so to not cause performance slowdowns, such as [here](https://github.com/ethereumjs/ethereumjs-monorepo/blob/d6aebec2ab90f6498c03a8905e56e2a779eba175/packages/blockchain/src/blockchain.ts#L330).

**Comment by jochem-brouwer** [comment]
Could you update this issue to write down what the uniform standard of debug is? I.e. the case casing, and how methods/objects are logged? (Seems like we now have an underscore thing like `find_path`? And it is all lowercase?)

**Comment by scorbajio** [comment]
> Could you update this issue to write down what the uniform standard of debug is? I.e. the case casing, and how methods/objects are logged? (Seems like we now have an underscore thing like `find_path`? And it is all lowercase?)

A brief resource on naming conventions can be found at ["Programming Naming Conventions Explained"](https://www.freecodecamp.org/news/programming-naming-conventions-explained/).
I've left some things open-ended for now (will do my best to note), but these are the rules I've implemented so far:
* Casings should always be lower case.
* Where necessary, snake case can be used to name either methods or objects or other logable things/events. I found a mix of this in the `trie` package, for example, where methods like `put`, `get`, `del`, and `find_path` are tagged using snake case, and objects like `extension_node` and `raw_node` as well as even specific conditions such as root values `before`/`after` reverts are tagged.
* The `trie` package also appends the branch index of nodes looked up in it's `onFound` callback, so use of numbers is also currently fair-use, but I'd say we can change this in a followup to #3692 since it would be more fit to include the branch index in the debug string, unless there are other comments on this.

We can also potentially keep this thread open to evolve our logging standards and naming here.
